packages/automation-contracts/scheduler/contracts/VestingSchedulerV2.sol

```diff
@@ -394,9 +666,20 @@ contract VestingSchedulerV2 is IVestingSchedulerV2, SuperAppBase {
         bytes32 configHash = keccak256(abi.encodePacked(superToken, sender, receiver));
         VestingSchedule memory schedule = vestingSchedules[configHash];
 
-        if (schedule.cliffAndFlowDate > block.timestamp ||
-            schedule.cliffAndFlowDate + START_DATE_VALID_AFTER < block.timestamp
-        ) revert TimeWindowInvalid();
+        if (schedule.isClaimable) {
+            // Ensure that the caller is the sender or the receiver if the vesting schedule requires claiming.
+            if (msg.sender != sender && msg.sender != receiver) {
+                revert CannotClaimFlowOnBehalf();
+            }
+            // Ensure that that the claming date is after the cliff/flow date and before the early end of the schedule
+            if (schedule.cliffAndFlowDate > block.timestamp ||
+                schedule.claimValidityDate < block.timestamp
+            ) revert TimeWindowInvalid();
+        } else {
+            if (schedule.cliffAndFlowDate > block.timestamp ||
+                schedule.cliffAndFlowDate + START_DATE_VALID_AFTER < block.timestamp
```

Comment:
There can probably be converge on something like this:

`relevantDate = isClaimable ? schedule.claimValidityDate : schedule.cliffAndFlowDate + START_DATE_VALID_AFTER`

Reply:
agreed and updated.